chain/starknet/src/chain.rs

```diff
@@ -0,0 +1,439 @@
+    async fn block_ptr_for_number(
+        &self,
+        logger: &Logger,
+        endpoint: &Arc<FirehoseEndpoint>,
+        number: BlockNumber,
+    ) -> Result<BlockPtr, Error> {
+        endpoint
+            .block_ptr_for_number::<codec::Block>(logger, number)
+            .await
+    }
+
+    /// Returns the closest final block ptr to the block ptr received.
+    /// On probablitics chain like Ethereum, final is determined by
+    /// the confirmations threshold configured for the Firehose stack (currently
+    /// hard-coded to 200).
+    ///
+    /// On some other chain like NEAR, the actual final block number is determined
+    /// from the block itself since it contains information about which block number
+    /// is final against the current block.
+    ///
+    /// To take an example, assuming we are on Ethereum, the final block pointer
+    /// for block #10212 would be the determined final block #10012 (10212 - 200 = 10012).
+    async fn final_block_ptr_for(
+        &self,
+        logger: &Logger,
+        endpoint: &Arc<FirehoseEndpoint>,
+        block: &codec::Block,
+    ) -> Result<BlockPtr, Error> {
+        // Firehose for Starknet has an hard-coded confirmations for finality sets to 100 block
+        // behind the current block. The magic value 100 here comes from this hard-coded Firehose
+        // value.
+        let final_block_number = match block.number() {
+            x if x >= 100 => x - 100,
+            _ => 0,
+        };
+
+        self.block_ptr_for_number(logger, endpoint, final_block_number)
+            .await
+    }
+}
+
+#[async_trait]
+impl TriggersAdapterTrait<Chain> for TriggersAdapter {
+    // Return the block that is `offset` blocks before the block pointed to
+    // by `ptr` from the local cache. An offset of 0 means the block itself,
+    // an offset of 1 means the block's parent etc. If the block is not in
+    // the local cache, return `None`
+    async fn ancestor_block(
+        &self,
+        _ptr: BlockPtr,
+        _offset: BlockNumber,
+    ) -> Result<Option<codec::Block>, Error> {
+        panic!("Should never be called since FirehoseBlockStream cannot resolve it")
+    }
+
+    // Returns a sequence of blocks in increasing order of block number.
+    // Each block will include all of its triggers that match the given `filter`.
+    // The sequence may omit blocks that contain no triggers,
+    // but all returned blocks must part of a same chain starting at `chain_base`.
+    // At least one block will be returned, even if it contains no triggers.
+    // `step_size` is the suggested number blocks to be scanned.
+    async fn scan_triggers(
+        &self,
+        _from: BlockNumber,
+        _to: BlockNumber,
+        _filter: &crate::adapter::TriggerFilter,
+    ) -> Result<Vec<BlockWithTriggers<Chain>>, Error> {
+        panic!("Should never be called since not used by FirehoseBlockStream")
+    }
+
+    // Used for reprocessing blocks when creating a data source.
```

Comment:
I think this comment is mislead, perhaps taken from runner.rs, where this is somewhat true but more widely this is used to transform the Block data into chain-specific triggers.

Reply:
Removed.